src/main.rs

```diff
@@ -128,13 +128,15 @@ impl Eq for Line {}
 
 impl PartialOrd for Line {
     fn partial_cmp(&self, other: &Line) -> Option<Ordering> {
-        self.score.partial_cmp(&other.score)
+        Some(self.cmp(other))
     }
 }
 
 impl Ord for Line {
     fn cmp(&self, other: &Self) -> Ordering {
-        self.score.cmp(&other.score)
+        self.score
+            .cmp(&other.score)
+            .then_with(|| other.path.cmp(&self.path))
```

Comment:
This is a nit, but thinking about this, I think we should probably use [`OsStr::from_bytes`](https://doc.rust-lang.org/std/ffi/struct.OsStr.html#method.from_bytes) on both paths here before comparing them, so that we get a string sort rather than a byte sort.

Reply:
Hmm, looking at `from_bytes`' implementation, it uses [`mem::transmute`](https://doc.rust-lang.org/std/mem/fn.transmute.html), which, if I'm understanding correctly, is a copy, which is a bit worrying. Is there no way to avoid the copy, by somehow "looking" at the slice of bytes as a string? If not, then perhaps `from_bytes` should be done once, when the input is consumed?

Reply:
Ah, no, it is transmuting the _reference_, not the bytes itself. Transmute is entirely a type-level operation, it does not do any copying.

Reply:
Ah, totally misread the "It's equivalent to C's `memcpy` under the hood", done!